worker/uniter/resolver_test.go

```diff
@@ -59,23 +60,35 @@ type conflictedResolverSuite struct {
 	baseResolverSuite
 }
 
+type rebootResolverSuite struct {
+	baseResolverSuite
+}
+
 var _ = gc.Suite(&caasResolverSuite{})
 var _ = gc.Suite(&iaasResolverSuite{})
 var _ = gc.Suite(&conflictedResolverSuite{})
+var _ = gc.Suite(&rebootResolverSuite{})
+
+const rebootNotDetected = false
+const rebootDetected = true
 
 func (s *caasResolverSuite) SetUpTest(c *gc.C) {
-	s.resolverSuite.SetUpTest(c, model.CAAS)
+	s.resolverSuite.SetUpTest(c, model.CAAS, rebootNotDetected)
 }
 
 func (s *iaasResolverSuite) SetUpTest(c *gc.C) {
-	s.resolverSuite.SetUpTest(c, model.IAAS)
+	s.resolverSuite.SetUpTest(c, model.IAAS, rebootNotDetected)
 }
 
 func (s *conflictedResolverSuite) SetUpTest(_ *gc.C) {
 	// NoOp, required to not panic.
 }
 
-func (s *baseResolverSuite) SetUpTest(c *gc.C, modelType model.ModelType) {
+func (s *rebootResolverSuite) SetUpTest(_ *gc.C) {
+	// NoOp, required to not panic.
```

Comment:
This seems weird, why is the SetUpTest required?

Reply:
I believe this is because the `SetUpTest` signature for the baseResolverSuite (https://github.com/juju/juju/blob/develop/worker/uniter/resolver_test.go#L78) does not adhere to the one expected by gocheck so it complains and throws a panic unless we override it for each suite that embeds the baseResolverSuite.